### PR TITLE
docs: add guide for investigating performance anomalies

### PR DESCRIPTION
**Pull request by ajassani**
## Summary

Adds a TraceLens conceptual notebook (`docs/conceptual/anomaly_detection_guide.ipynb`) that walks through detecting and profiling occasional slow steps without paying the cost of always saving traces.

The pattern: run the PyTorch profiler continuously with CUDA-only activities, configure the schedule so the buffer flushes on every `prof.step()`, and let a custom trace handler persist only the traces for steps whose duration exceeds a threshold. Healthy steps' events are flushed and discarded; anomalous steps produce small single-step traces ready for downstream analysis.

The notebook covers:
- ResNet18 example with a synthetic anomaly injector (occasional large matmul)
- Baseline duration measurement and a diagnostic per-step timeline plot
- Profiler setup with `schedule(wait=0, warmup=0, active=1)` for per-step buffer flush
- Trace handler with a FIFO queue to keep `(step, duration)` ordering correct
- Always-on profiling loop with a generic body — anomalies surface only through the saved trace files
- TraceLens perf-report (`gpu_timeline`, `kernel_summary`) generated on one of the saved anomaly traces

## Test plan

- [x] Notebook validated end-to-end on an MI300X GPU node (200 steps, 14 anomalies detected and saved)
- [x] Diagnostic plot renders with threshold line + above-threshold markers
- [x] `gpu_timeline` and `kernel_summary` DataFrames render as HTML tables in the perf-report cells
- [x] Training loop output is generic (no `ANOMALY` markers, no threshold leak) — anomaly steps are reported only by the trace handler's `saved: ...` lines
- [x] JSON validity check passes

### CODE REVIEW

File: docs/conceptual/anomaly_detection_guide.ipynb [hunk 1]
```diff
@@ -0,0 +1,848 @@
...
...RCCCHEBaVv7gQIIYQQQjRGWloaU6ZM4fXXX+eGG26gqKiILVu2cMcdd5CUlERhYSGLFy8GwNfXF7PZzJgxY4iKimLLli3o9Xpefvllxo4dy6FDhzAajQBs3LgRZ2dnNm/eTEJCAnfeeSd+fn688sorzXm6QgghhLgAJAgihBBCiFYpLS0Ni8XCjTfeSHh4OAC9evUCwMXFhYqKCoKDg6uPX7p0KTabjc8//xyNRgPA4sWL8fb2ZvPmzYwePRoAo9HIokWLcHV1pUePHrz44os89dRTvPTSS2i1MohWCCGEaM3kTi6EEEKIVqlPnz6MGjWKXr16cfPNN/PZZ5+Rl5dX5/EHDx4kLi4ODw8P3N3dcXd3x9fXl/LycuLj42t8r6ura/W/o6KiKC4uJjk5uUnPRwghhBBNT0aCCCGEEKJV0ul0rF+/nu3bt/P777/z/vvv8+yzzxIdHV3r8cXFxQwYMIBvvvnmnPcCAgKaOrlCCCGEaAEkCCKEEEKIVkuj0TB06FCGDh3K888/T3h4OKtXr8ZoNGK1Wmsc279/f5YvX05gYCCenp51fufBgwcpKyvDxcUFgJ07d+Lu7k67du2a9FyEEEII0fRkOowQQgghWqXo6GheffVV9uzZQ1JSEqtWrSIrK4tu3boRERHBoUOHiImJITs7G7PZzNSpU/H39+f6669ny5YtnD59ms2bN/Pwww+TkpJS/b0mk4mZM2dy7NgxfvnlF1544QUefPBBWQ9ECCGEcAAyEkQIIYQQrZKnpyd//fUX77zzDoWFhYSHh/PWW28xbtw4LrvsMjZv3sxll11GcXExmzZtYsSIEfz11188/fTT3HjjjRQVFdGmTRtGjRpVY2TIqFGj6Ny5M8OHD6eiooIpU6Ywb9685jtRIYQQQlwwGkVRlOZOhBBCCCFESzBjxgzy8/NZs2ZNcydFCCGEEE1AxnUKIYQQQgghhBDikiBBECGEEEIIIYQQQlwSZDqMEEIIIYQQQgghLgkyEkQIIYQQQgghhBCXBAmCCCGEEEIIIYQQ4pIgQRAhhBBCCCGEEEJcEiQIIoQQQgghhBBCiEuCBEGEEEIIIYQQQghxSZAgiBBCCCGEEEIIIS4JEgQRQgghhBBCCCHEJUGCIEIIIYQQQgghhLgkSBBECCGEEEIIIYQQl4T/B8CsiTmptDbRAAAAAElFTkSuQmCC",
```
Comment (gabeweisz) [inline docs/conceptual/anomaly_detection_guide.ipynb:146]: Is the data important to have here or can we clear all cells?

**Review by gabeweisz — APPROVED** [review]

File: docs/conceptual/anomaly_detection_guide.ipynb (hunk 1 above)
Comment (ajassani) [reply docs/conceptual/anomaly_detection_guide.ipynb:146]: data is important to give the reader context
